### PR TITLE
Feature/support conan package creation from local build

### PR DESCRIPTION
**Pull request by maxfaine**
Feature/support conan package creation from local build

### CODE REVIEW

File: conanfile.py [hunk 1]
```diff
@@ -31,7 +30,7 @@ class UpCpp(ConanFile):
         "shared": False,
         "fPIC": False,
         "build_testing": False,
-        "build_unbundled": False,
+        "build_unbundled": True,
```
Comment (gregmedd) [inline conanfile.py:33]: Why is it important to change the default? Does that change the build instructions in the README file?

File: conanfile.py [hunk 2]
```diff
@@ -49,13 +48,34 @@ def generate(self):
         tc.generate()
 
     def build(self):
+        if os.environ.get("BUILD_DIR_FULL_PATH") is not None:
+            return
```
Comment (gregmedd) [inline conanfile.py:52]: Do we have documentation for all of the variables (both environment and command line) that are taken by the build scripts? This should be added there with an explanation of what it is, what it does, and how to set it.

File: conanfile.py [hunk 3]
```diff
@@ -49,13 +48,34 @@ def generate(self):
...
+            self.cpp_info.libs = collect_libs(self)
+
+            self.cpp_info.set_property("cmake_target_name", "up-cpp::up-cpp")
+            self.cpp_info.requires = ["spdlog::spdlog", "protobuf::protobuf"]
+
+            self.cpp_info.names["cmake_find_package"] = "up-cpp"
+            self.cpp_info.names["cmake_find_package_multi"] = "up-cpp"
+
+        else:
+            self.cpp_info.libs = ["up-cpp"]
```
Comment (gregmedd) [inline conanfile.py:81]: These are fairly significant changes. Can you provide an explanation of what these changes are and what they do?

**Review by gregmedd — CHANGES_REQUESTED** [review]
I was able to build this library on a clean ubuntu 22.04 system by following the README. However, I was not able to subsequently build up-client-zenoh-cpp against it. Please make sure that it is possible to go from zenoh-c to up-cpp, up-client-zenoh-cpp, and finally up-zenoh-example-cpp with these changes in place.

Your CI failure is related to the update in the README - since `version = "0.1.1-dev"` was removed from conanfile.py, you will need to specify a version in [.github/workflows/ci.yml](https://github.com/eclipse-uprotocol/up-cpp/blob/main/.github/workflows/ci.yml).

Lastly, I'd like for you to update both the commits and this review with a meaningful description of what these changes are and what they do. The commits can be squashed and reworded however you best feel conveys the content of your work using git rebase -i ....

**Comment by stevenhartley** [comment]
> I was able to build this library on a clean ubuntu 22.04 system by following the README. However, I was not able to subsequently build up-client-zenoh-cpp against it. Please make sure that it is possible to go from zenoh-c to up-cpp, up-client-zenoh-cpp, and finally up-zenoh-example-cpp with these changes in place.
> 
> Your CI failure is related to the update in the README - since `version = "0.1.1-dev"` was removed from conanfile.py, you will need to specify a version in [.github/workflows/ci.yml](https://github.com/eclipse-uprotocol/up-cpp/blob/main/.github/workflows/ci.yml).
> 
> Lastly, I'd like for you to update both the commits and this review with a meaningful description of what these changes are and what they do. The commits can be squashed and reworded however you best feel conveys the content of your work using git rebase -i ....

We should probably not specify version in the ci workflow as this is difficult to maintain but leverage conan to build and pull the right dependencies. Maybe git submodules or we push to conan central.

**Comment by gregmedd** [comment]
Closing stale PRs. We are tracking conan recipes on the project board.

**Pull request closed without merging**